net/ipforward/ipforward.h

```diff
@@ -42,6 +42,16 @@
 #  define CONFIG_NET_IPFORWARD_NSTRUCT 4
 #endif
 
+#if CONFIG_IOB_NBUFFERS <= CONFIG_NET_IPFORWARD_NSTRUCT
+#  error CONFIG_IOB_NBUFFERS <= CONFIG_NET_IPFORWARD_NSTRUCT \
+         IP forward may consume all the IOB and break netdev logic.
+#endif
```

Comment:
```suggestion
static_assert(CONFIG_IOB_NBUFFERS > CONFIG_NET_IPFORWARD_NSTRUCT,
              "IP forward may consume all the IOB and break netdev logic");
```
nit: How about using a `static_assert` instead? It is defined in `assert.h` and seems cleaner.

Reply:
Unfortunately I am not aware of a compile-time verification that could raise a warning.

> The below condition makes a warning because it won't let NuttX work totally wrong, but may only waste some resources under some conditions.

If the relevance of the warning is questionable, maybe we should just delete it. Our CI runs with `-Werror`, so if some defconfig purposely defines a set of values (which are valid) that would trigger the warning, then the CI would raise a false positive here.

Reply:
> If the relevance of the warning is questionable, maybe we should just delete it. Our CI runs with `-Werror`, so if some defconfig purposely defines a set of values (which are valid) that would trigger the warning, then the CI would raise a false positive here.

OK, it's not a very important check so I removed it: if it fails, NuttX will not work under best performance but still works well without problems.